device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/https/HttpsConnection.java

```diff
@@ -101,6 +101,16 @@ public HttpsConnection(URL url, HttpsMethod method, final ProxySettings proxySet
             throw new IllegalArgumentException(errMsg);
         }
 
+        String host = url.getHost();
+        if (!isHttps && !host.equalsIgnoreCase("localhost"))
+        {
+            // Currently, IoT Edge runtime can dictate that this client library sends an HTTP request, but it will
+            // only ever be to localhost. The Edge team is currently working to remove this requirement. Eventually,
+            // the edge runtime will just make this client use unix domain sockets instead.
+            // Once that happens, we can remove support for HTTP requests entirely.
+            throw new IllegalArgumentException("Cannot do HTTP requests to any host other than localhost");
```

Comment:
Is this worth bringing up as a breaking change with Mark? My guess is there is no way a customer should run into this, therefore it isn't really breaking. Can they?

Reply:
As long as the edge workload URI always points to localhost, which the edge team confirmed, then this change shouldn't be noticed by anyone. There are no other codepaths that made this SDK send an HTTP request